test/files/run/productElementName.scala

```diff
@@ -0,0 +1,11 @@
+
+case class User(name: String, age: Int)
+
+object Test extends App {
+  def pretty(p: Product): String =
+    p.productElementNames.zip(p.productIterator)
+     .map { case (name, value) => s"$name=$value" }
+     .mkString(p.productPrefix + "(", ", ", ")")
+  println(pretty(User("Susan", 42)))
+}
```

Comment:
We should also test with:
 - [ ] non alphanumeric names (is the `name.decoded` done before code gen?)
 - [ ] case classes with non-public elements (just in case there is any problems in the current implementation related to [these comments](https://github.com/scala/scala/blob/83890fc1362aa126f4b349080cdcb8c83817dc00/src/reflect/scala/reflect/internal/Symbols.scala#L2072-L2087))
 - [ ] case classes with secondary parameter lists or with secondary constructors (again, to make sure we don't have latent bugs in `product*` code gen)

Reply:
Good idea